enterprise/causal-clustering/src/main/java/org/neo4j/causalclustering/catchup/storecopy/GetStoreRequestHandler.java

```diff
@@ -84,13 +83,58 @@ private void sendFiles( ChannelHandlerContext ctx ) throws IOException
         }
     }
 
-    private void sendFile( ChannelHandlerContext ctx, File file ) throws FileNotFoundException
+    private void sendFile( ChannelHandlerContext ctx, File file ) throws IOException
     {
         ctx.writeAndFlush( ResponseMessageType.FILE );
-        long initialLength = file.length();
-        ctx.writeAndFlush( new FileHeader( file.getName(), initialLength ) );
-        ctx.writeAndFlush( new ChunkedNioStream( new LimitedLengthReadableByteChannel(
-                new FileInputStream( file ).getChannel(), initialLength ) ) );
+        ctx.writeAndFlush( new FileHeader( file.getName() ) );
+        ByteBuffer buffer = ByteBuffer.allocate( FileChunk.DEFAULT_SIZE );
```

Comment:
If you use `allocateDirect` there will be one less memory copy in `channel.read`.